### ISSUE TITLE
Not working on catalina public beta 3

### ISSUE THREAD

**Issue by nicolapietropaolodeagostini** [toland/qlmarkdown#89]
This and other QL plugins are no longer recognized as created by an identified developer:
"QLMarkdown.qlgenerator" was blocked from opening because it is not from an identified developer.

But I'm even not able to open it anyway for some reason.

**Comment by danielcompton** [comment]
I suspect this might also need to be [notarized](https://developer.apple.com/documentation/security/notarizing_your_app_before_distribution).

**Comment by adityasrini** [comment]
Not working on GA Catalina.

**Comment by tfflyer** [comment]
怎么办呢怎么办，大佬们，macos15用不了啊

**Comment by zhaoyaqiong** [comment]
Not working on 10.15 (19A583)

**Comment by stefanwascoding** [comment]
After modifying the source as described in https://github.com/anthonygelibert/QLColorCode/issues/51#issuecomment-541228682 I can use the resulting qlgenerator without problem on Catalina.

[catalina.patch.txt](https://github.com/toland/qlmarkdown/files/3746764/catalina.patch.txt)

**Comment by tthkbw** [comment]
Thank you! This worked perfectly for me and I can now quicklook markdown again.

**Comment by diliedevs** [comment]
Hi, just installed the latest release, but it's still not working on Catalina 10.15  (19A602)

**Comment by tattali** [comment]
@diliedevs Weird, it no longer works for me either. Can you try with this one? It works for me even after qlmanage reset and computer restart. But I want be sure thats not only work on my mac.

[QLMarkdown.qlgenerator.zip](https://github.com/toland/qlmarkdown/files/3775741/QLMarkdown.qlgenerator.zip)

**Comment by diliedevs** [comment]
Yes, the new one works! Thanks for fixing this... having trouble with all my quicklook plugins. Yours is only the second one to be fixed. Thanks!!

**Comment by tattali** [comment]
Thanks to @stefanwascoding answer

**Comment by bnussbau** [comment]
FYI: I had to download the release with Safari, with Chrome it wouldn't work on Catalina.
You can check using `codesign -vvv --deep --strict /path/to/QLMarkdown.qlgenerator` if the binary is valid. It should print out
```
QLMarkdown.qlgenerator: valid on disk
QLMarkdown.qlgenerator: satisfies its Designated Requirement
```

When downloading with Chrome i had an `QLMarkdown.qlgenerator: unsealed contents present in the bundle root` error.

**Comment by tattali** [comment]
It seems thats the quick look should be notarized with 99$ fee. 
To prevent this you should :
```
$ open ~/Library/QuickLook/QLMarkdown.qlgenerator/Contents/MacOS/
```
right click open on the binary
then in the Preferences panel > go to Security & Privacy > and click on Allow QLMarkdown
then if needed re run qlmanage
```
qlmanage -r
```
If you find an other way please share it

**Comment by sbarex** [comment]
Has anyone tried to see if it works on 10.15.1?
I updated my virtual machine and it seems that now the various qlgenerators works but I'm not sure if I did something strange in the settings or if it's a problem solved in the last update.

**Comment by stefanwascoding** [comment]
> Has anyone tried to see if it works on 10.15.1?
@sbarex Updated macOS ... QLMarkdown (compiled from source) is still running.

**Comment by parleer** [comment]
Got it working on MacOS Catalina (10.15) with the patch provided above. 

Here's an exact recipe for those interested:

```
# Get source code
git clone https://github.com/toland/qlmarkdown
cd qlmarkdown

# Update to latest discount library (Per QLMarkdown's Readme.markdown guide)
./discount-config/update-discount.sh

# Install patch for MacOS Catalina
wget -O https://github.com/toland/qlmarkdown/files/3746764/catalina.patch.txt
git apply catalina.patch.txt

# Build and install
xcrun xcodebuild
cp -R ./build/Release/QLMarkdown.qlgenerator ~/Library/QuickLook/

# Reset QuickLook cache and relaunch Finder
qlmanage -r
sudo killall finder
```

**Comment by Neberheim** [comment]
@parleer running `wget` as written returned a "missing URL" error. Worked around it by downloading through browser and placing it manually. I was also not able to copy the directory using `cp` and had to drag & drop through the Finder, strangely. Also `Finder` is the correct case for the process. Got it up and running with the workarounds though, thanks!

**Comment by parleer** [comment]
> @parleer running `wget` as written returned a "missing URL" error. Worked around it by downloading through browser and placing it manually. I was also not able to copy the directory using `cp` and had to drag & drop through the Finder, strangely. Also `Finder` is the correct case for the process. Got it up and running with the workarounds though, thanks!

@Neberheim Sorry, I wast trying to re-create my steps and missed a few things. I've updated the script and it should work now. Not sure why you were running into trouble with `cp` though.

**Comment by Neberheim** [comment]
@parleer It's a permissions issue it turns out, but giving iTerm full disk access in the Privacy system preference and doing the same with Terminal still yielded no results, even with sudo. Interestingly, while no exit message was returned with qlmarkdown as the working directory, sending it from my home directory returned `cp: /Library/QuickLook/QLMarkdown.qlgenerator: Permission denied`

So strangely `~ % sudo cp -R ~/qlmarkdown/build/Release/QLMarkdown.qlgenerator [...]` ended up working properly, even though `qlmarkdown % sudo cp [...]` did not.

Since it's outside the purview of this thread, I'll leave it at that and investigate elsewhere.

**Comment by jackStalnaker** [comment]
You can get the version in homebrew to work by removing the quarantine attribute from the cask file like this: 
`xattr -d com.apple.quarantine /usr/local/Caskroom/qlmarkdown/1.3.6/QLMarkdown.qlgenerator`

**Comment by brad-provident** [comment]
I get this error after trying to install the patch:

`xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun`

**Comment by parleer** [comment]
@brad-provident Run `xcode-select --install` to install Command Line Tools

**Comment by tattali** [comment]
Hi, [here](https://github.com/whomwah/qlstephen/issues/81#issuecomment-572930525) is the most common solution.
Before doing this, make sure that the plug-in is installed and that you have tried to preview a valid markdown file.

**Comment by relikd** [comment]
or as noted later in the thread, run `xattr -cr ~/Library/QuickLook/QLMarkdown.qlgenerator`. Though it should be enough to run it on the zip file before extracting.

You can even create an Automator action and add it to your right click menu, e.g. "remove extended attributes". ;)

![screen](https://user-images.githubusercontent.com/1321720/72518353-6ec4fa80-3855-11ea-872b-a8ecdc5daa07.png)
